src/metabase/api/public.clj

```diff
@@ -113,7 +114,7 @@
                          (qp.streaming/streaming-response [{:keys [reducedf], :as context} export-format]
                            (let [context  (assoc context :reducedf (public-reducedf reducedf))
                                  in-chan  (binding [api/*current-user-permissions-set* (atom #{"/"})]
-                                            (qp/process-query-and-save-execution! query info context))
+                                            ((or qp-runner qp/process-query-and-save-execution!) query info context))
```

Comment:
seeing how this is effectively an optional param  I think this is a key indicator that `qp-runner` should be a keyword param and not positional... you can change `& options` to 

```clj
& {:keys [qp-runner], :or {qp-runner qp/process-query-and-save-execution!}, :as options}
```

and use `m/mapply` instead of `apply` to pass the options on to the next function